App.tsx

```diff
@@ -1,12 +1,154 @@
-import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, Text, View } from 'react-native';
+import { StyleSheet } from 'react-native';
+import { ItemList } from './ItemList';
+import { ImageModal } from './ImageModal';
+import { DetailsPage } from './DetailsPage';
+import { createStackNavigator } from '@react-navigation/stack';
+import { NavigationContainer } from '@react-navigation/native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { WeatherApp } from './WeatherApp/Weather';
+import { WeatherStackParam } from './WeatherApp/WeatherStackParam';
+import { ScannerScreen } from './BarCodeScanner/BarCodeScannerScreen';
+import { ScannerStackParam } from './BarCodeScanner/ScannerStackParam';
+import { ProductDetails } from './BarCodeScanner/ProductDetails';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import FavoritesPage from './BarCodeScanner/FavoritesPage';
+import { Ionicons } from '@expo/vector-icons';
+
+export type StackParamList = {
+  Gallery: undefined,
+  Details: { id: number; url: string},
+  Modal: {id: number; url: string},
+}
+
+const Stack = createStackNavigator();
+const Weather = createStackNavigator<WeatherStackParam>();
+const Scanner = createBottomTabNavigator<ScannerStackParam>();
+const Drawer = createDrawerNavigator();
+const Details = createStackNavigator();
+
+const WeatherNav = () => {
+  return (
+    <Weather.Navigator>
+      <Weather.Screen name="main" component={WeatherApp}/>
+    </Weather.Navigator>
+  )
+}
+
+const Tab = createBottomTabNavigator();
+
+const ProductDetailsStack = () => (
+  <Stack.Navigator>
+    <Stack.Screen name="ProductDetails" component={ProductDetails} />
+  </Stack.Navigator>
+);
+
+const ScannerNav = () => {
+  return (
+    <Scanner.Navigator>
+      <Scanner.Screen 
+      name="main" 
+      component={ScannerScreen}
```

Comment:
This component should be a `ScannerStack` which would consist of the product details screen and the Scanner screen. I'd update your `ProductDetailsStack` to what I've shared below. This will hide the extra tab at the bottom of the screen and make it easier to navigate to product details.

```
const ScannerStackScreeen = () => (
  <Stack.Navigator>
    <Stack.Screen name="ScannerScreen" component={ScannerScreen} />
    <Stack.Screen name="ProductDetails" component={ProductDetails} />
  </Stack.Navigator>
);


```